solutions/cses-1745.mdx

```diff
@@ -0,0 +1,133 @@
+---
+id: cses-1745
+source: CSES
+title: Money Sums
+author: Sofia Yang
+---
+
+[Codeforces Editorial](https://codeforces.com/blog/entry/70018)
+
+## Solution
+
+**Time Complexity:** $\mathcal{O}(N\cdot X)$
+
+## Implementation
+
+
+<LanguageSection>
+
+
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class cses1745 {
+public static int maxSum = (int) 1e5;
+
+	public static void main(String[] args) throws IOException {
+		Kattio k = new Kattio(System.in);
+
+		int N = k.getInt();
+
+		int[] value = new int[N + 1];
+		for (int i = 1; i <= N; i++) {
+			value[i] = k.getInt();
+		}
+
+		// DP[i][j] is true, if it is possible to make a sum of j with i coins.
+		boolean[][] dp = new boolean[N + 1][maxSum + 1];
+
+		dp[0][0] = true;
+
+		/**
+		 * Loop through all coins and all possible sums.
+		 * 1. If it is possible to make the sum with less than i coins,
+		 * Then DP[i][sum] will be true.
+		 * 2. If it is possible to make a sum of (curSum - value[i])
+		 * DP[curSum] will also be true.
+		 */
+		for (int i = 1; i <= N; i++) {
+			for (int curSum = 0; curSum <= maxSum; curSum++) {
+				dp[i][curSum] = dp[i - 1][curSum];
+				int prevSum = curSum - value[i];
+				if (prevSum >= 0 && dp[i - 1][prevSum]) {
+					dp[i][curSum] = true;
+				}
+			}
+		}
+
+		// Check the number of possible sums.
+		ArrayList<Integer> possible = new ArrayList<>();
+		for (int sum = 1; sum <= maxSum; sum++) {
+			if (dp[N][sum]) {
+				possible.add(sum);
+			}
+		}
+		System.out.println(possible.size());
+		for (int i : possible) {
+			System.out.print(i + " ");
+		}
+
+	}
+
+	static class Kattio extends PrintWriter {
```

Comment:
use kattio codesnip

Reply:
I did (I think)
I got it from here:
https://open.kattis.com/download/Kattio.java

Reply:
Sorry, I'm dumb.
It's this one?
https://usaco.guide/general/working-mdx?lang=java#kattio

Reply:
yea, replace the kattio class with `// CodeSnip{Kattio}`

Reply:
oh, your kattio thing is a bit different